### DISCUSSION TITLE
Negative baseline & high contributions outputs

### DISCUSSION THREAD

**Discussion by isaacmalka** [google/meridian#341] [Q&A]
Hi everyone,

We’re currently testing potential future transitioning from LightweightMMM to Meridian, where we didn’t encounter negative outputs for baseline values. In Meridian, however, we’re seeing negative baseline estimates for some media channels, which may be causing contributions for certain channels to exceed 100%.

Is there a specific hyperparameter we can tune to prevent negative baseline values? I assumed these values might be sampled from a half-normal distribution, but I’m not certain.

Any insights or guidance would be greatly appreciated. Thank you!

**Comment by jyklly** [comment]
Please see the advice given on https://developers.google.com/meridian/docs/advanced-modeling/model-debugging#baseline-low-or-negative. If you opt to using the [total media contribution prior](https://developers.google.com/meridian/docs/advanced-modeling/unknown-revenue-kpi#set-total-media-contribution-prior) you can set a p_mean or p_sd value that is appropriate for your use-case and may help resolve the negative baseline issue.

**Reply by isaacmalka** [reply]
Thanks Joseph. I will try it out, this solution makes sense. We're currently testing on a national level (with no reach, frequency nor impressions). Is that a reason to be concerned wtr to the model output or should it be able to be at least as performative as LightWeightMMM?

**Reply by isaacmalka** [reply]
The baseline issue was resolved; but the highest contributing are almost twice as high compared to when using LightWeightMMM. Is there a way to infer way and mitigate that?

**Comment by mikewurm** [comment]
If you have a strong prior belief about the ROI or contribution, then it may help to use a stronger prior distribution with lower variance. Further advice is [available here](https://developers.google.com/meridian/docs/advanced-modeling/roi-priors-and-calibration?hl=en).

**Reply by isaacmalka** [reply]
Thanks @mikewurm - thing is, we have found that some channels now exhibit a much higher contribution compared to what we got with LightWeightMMM. What's the best approach here in your opinion? Can we incorporate prior knowledge per channel or overall like we currently implemented? (like 50% contribution mean, ~0.1 STD.. but this is currently implemeted across all channels).

**Reply by mikewurm** [reply]
Either approach could be reasonable. If you have prior knowledge specific to certain channels, it may be best to use this.

Although LightWeightMMM and Meridian have a lot of similarities, they are different models and will not necessarily give the same results. Meridian is a customizable modeling framework, and different modeling choices can yield different results.

**Comment by wz-jellyfish** [comment]
I noticed that the instructions for deploying the “Custom total paid media contribution prior” (as described in [this documentation](https://developers.google.com/meridian/docs/advanced-modeling/unknown-revenue-kpi#set-total-media-contribution-prior)) might be misleading when the KPI is not revenue. After testing a model with a non-revenue KPI, I encountered a negative value for lognormal_mu.

**Reply by mikewurm** [reply]
A negative value for lognormal_mu is not necessarily problematic. A negative value means that the median of the ROI prior distribution is less than one. When KPI is not revenue, ROI is defined as incremental KPI units per monetary unit spent, so a value less than one could make sense.

Could you be more specific about which part you feel is misleading? Thanks!

**Reply by wz-jellyfish** [reply]
if the KPI is not revenue but the model also include revenue_per_kpi, should we use KPI * revenue_per_kpi as the input to generate the default prior distribution?

**Reply by mikewurm** [reply]
If `revenue_per_kpi` is included, then in code snippet you should replace `kpi` (which represents total KPI) with total revenue.  In that case, `p_mean` is the prior mean percentage of *revenue* attributed to paid media.
